<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1000</width>
    <height>700</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <layout class="QGridLayout" name="gridLayout_5">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item row="1" column="0">
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>550</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>Tab 1</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QComboBox" name="filenameBox">
                <property name="maxVisibleItems">
                 <number>20</number>
                </property>
                <property name="maxCount">
                 <number>100</number>
                </property>
                <item>
                 <property name="text">
                  <string>0</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>2</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>3</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>4</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>5</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>6</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>7</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>8</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>9</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>10</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>11</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>12</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>13</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>14</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>15</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>16</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>17</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>18</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>19</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>20</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>21</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>22</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>23</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>24</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>25</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>26</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="uploadOriginImgButton">
                <property name="maximumSize">
                 <size>
                  <width>150</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="layoutDirection">
                 <enum>Qt::LeftToRight</enum>
                </property>
                <property name="text">
                 <string>Upload
 original image</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton">
                <property name="text">
                 <string>Blur
original image</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_4">
                <property name="text">
                 <string>Equalize Hist</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="uploadGrayImgButton">
                <property name="maximumSize">
                 <size>
                  <width>150</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Upload
 gray image</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="uploadBinaryImgButton">
                <property name="maximumSize">
                 <size>
                  <width>150</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Upload
binary image</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_3">
                <property name="text">
                 <string>Upload
morph image</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QGraphicsView" name="subView">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>250</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_4">
         <attribute name="title">
          <string>Tab 2</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QGraphicsView" name="graphicsView_3"/>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QToolBox" name="toolBox">
        <property name="maximumSize">
         <size>
          <width>400</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="page_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>-16</y>
           <width>383</width>
           <height>326</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Threshold and Morph filters</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Threshold</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="threshValue">
                <property name="minimumSize">
                 <size>
                  <width>16</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>0</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSlider" name="thresholdSlider">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>255</number>
                </property>
                <property name="value">
                 <number>140</number>
                </property>
                <property name="sliderPosition">
                 <number>140</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="tickPosition">
                 <enum>QSlider::TicksBothSides</enum>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_6">
              <item row="0" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <item>
                 <layout class="QGridLayout" name="gridLayout_8">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>Rows Structuring element</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QSpinBox" name="rowSpinBox">
                    <property name="maximumSize">
                     <size>
                      <width>50</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="minimum">
                     <number>1</number>
                    </property>
                    <property name="singleStep">
                     <number>2</number>
                    </property>
                    <property name="value">
                     <number>1</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QGridLayout" name="gridLayout_7">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_6">
                    <property name="text">
                     <string>Cols Structuring element</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QSpinBox" name="colSpinBox">
                    <property name="maximumSize">
                     <size>
                      <width>50</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="minimum">
                     <number>1</number>
                    </property>
                    <property name="singleStep">
                     <number>2</number>
                    </property>
                    <property name="value">
                     <number>7</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_10">
                <item>
                 <widget class="QComboBox" name="morphFilterType">
                  <property name="currentIndex">
                   <number>1</number>
                  </property>
                  <item>
                   <property name="text">
                    <string>Erosion</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Dilation</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Open</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Close</string>
                   </property>
                  </item>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="morphFilterButton">
                  <property name="text">
                   <string>Apply filter</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_2">
              <property name="text">
               <string>Reset imgGpuBinaryMorph</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout"/>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_12">
              <item row="0" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_12">
                <item>
                 <layout class="QGridLayout" name="gridLayout_13">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_9">
                    <property name="text">
                     <string>Rows Structuring element</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QSpinBox" name="rowSpinBox_3">
                    <property name="maximumSize">
                     <size>
                      <width>50</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="minimum">
                     <number>1</number>
                    </property>
                    <property name="singleStep">
                     <number>2</number>
                    </property>
                    <property name="value">
                     <number>31</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QGridLayout" name="gridLayout_14">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_10">
                    <property name="text">
                     <string>Cols Structuring element</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QSpinBox" name="colSpinBox_3">
                    <property name="maximumSize">
                     <size>
                      <width>50</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="minimum">
                     <number>1</number>
                    </property>
                    <property name="singleStep">
                     <number>2</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_13">
                <item>
                 <widget class="QComboBox" name="morphFilterType_3">
                  <item>
                   <property name="text">
                    <string>Erosion</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Dilation</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Open</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Close</string>
                   </property>
                  </item>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="morphFilterButton_3">
                  <property name="text">
                   <string>Apply filter</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="findEdges">
              <property name="text">
               <string>Find edges of numbers</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="applyAllAndFindEdges">
              <property name="text">
               <string>Apply all filters and find edges of numbers</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_4">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>400</width>
           <height>310</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Blur</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_16">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_15">
            <item row="1" column="0">
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Gaussian Blur
Kernel size</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="gausKernelSize">
                <property name="minimum">
                 <number>3</number>
                </property>
                <property name="maximum">
                 <number>31</number>
                </property>
                <property name="singleStep">
                 <number>2</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_5">
                <property name="text">
                 <string>Apply Blur</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_6">
                <property name="text">
                 <string>Reset</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget_2">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Main</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_23">
        <item row="0" column="0">
         <widget class="QGraphicsView" name="graphicsView">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>400</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Threshold</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_24">
        <item row="0" column="0">
         <widget class="QGraphicsView" name="thresholdView"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>MorphFilter</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_25">
        <item row="0" column="0">
         <widget class="QGraphicsView" name="morphView"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_6">
       <attribute name="title">
        <string>Edges</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_26">
        <item row="0" column="0">
         <widget class="QGraphicsView" name="edgesView"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_7">
       <attribute name="title">
        <string>ROI</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_9">
        <item row="0" column="0">
         <widget class="QGraphicsView" name="roiView"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuROI">
    <property name="title">
     <string>ROI</string>
    </property>
    <addaction name="actionSelect_Roi"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menuROI"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSelect_Roi">
   <property name="text">
    <string>Select ROI</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
